File: self_checkout_backend/models/history.js
const mongoose = require('mongoose');
const { Schema,model } = mongoose;

const historySchema = Schema({
    storeid:{
        type:String,
        required:true
    },
    storename:{
        type:String,
        required:true
    },
    orderid:{
        type:String,
        required:true
    },
    userid:{
        type:String,
        required:true
    },
    totalprice:{
        type:String,
        required:true
    },
    items:{
        type:Array,
        required:true
    }
},{timestamps:true});
const historyModel = model('History',historySchema,'history');
module.exports = historyModel; 

File: self_checkout_adminpanel/lib/models/history.dart
class History {
  final String id;
  final String storeId;
  final String storeName;
  final String orderId;
  final String userId;
  final double totalPrice;
  final DateTime createdAt;

  History({
    required this.id,
    required this.storeId,
    required this.storeName,
    required this.orderId,
    required this.userId,
    required this.totalPrice,
    required this.createdAt,
  });

  factory History.fromJson(Map<String, dynamic> json) {
    return History(
      id: json['_id'],
      storeId: json['storeid'],
      storeName: json['storename'],
      orderId: json['orderid'],
      userId: json['userid'],
      totalPrice: double.parse(json['totalprice']),
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}
